main.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+# This is a modified version of the Enumerable module.
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    if is_a? Array
+      min = 0
+      max = length
+      max.times do
+        yield(self[min])
+        min += 1
+      end
+    elsif is_a? Range
+      min = self.min
+      max = self.max
+      until min == max
+        p min
+        min += 1
+      end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    i = 0
+    if is_a? Array
+      min = 0
+      max = length
+      max.times do
+        yield(self[min], min)
+        min += 1
+        i += 1
+      end
+    elsif is_a? Range
+      min = self.min
+      max = self.max
+      until min == max
+        p min
+        min += 1
+      end
+    end
+end
+
+  def my_select
+    if block_given?
+      selection = []
+      my_each do |i|
+        selection << i if yield(i)
+      end
+      selection
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(arg = nil)
+    if arg.nil? && block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif arg.is_a? Regexp
+      my_each { |i| return false unless i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| return false unless i.is_a?(arg) }
+    elsif !block_given? || arg.nil?
+      my_each { |i| return false unless i.nil? || i == false }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if arg.nil? && block_given?
+      my_each { |i| return true if yield(i) }
+    elsif arg.is_a? Regexp
+      my_each { |i| return true if i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| return true if i.is_a?(arg) }
+    elsif !block_given? || arg.nil?
+      my_each { |i| return true unless i.nil? || i == false }
+    else
+      my_each { |i| return true if i.nil? || i == false }
+    end
+    false
+ end
```

Comment:
* [ ] `my_any?` is identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?('cat') == array.any?('cat') # false
  ```